src/app/components/PageLayout/ScrollingCard.tsx

```diff
@@ -0,0 +1,20 @@
+import Card from '@mui/material/Card'
+import { FC, useEffect, useRef } from 'react'
+import { useLocation } from 'react-router-dom'
+
+export const ScrollingCard: FC<Parameters<typeof Card>[0]> = props => {
+  const { id } = props
+  const { hash } = useLocation()
+  const cardRef = useRef<HTMLDivElement>(null)
+  const card = cardRef.current
+
+  useEffect(() => {
+    if (id && hash === `#${id}` && card) {
+      card.scrollIntoView({
+        behavior: 'smooth',
+        block: 'start',
+      })
+    }
+  }, [id, hash, card])
+  return <Card {...props} ref={cardRef} />
```

Comment:
Could this be a completely general abstraction, instead of tying it to Card?

Reply:
On some level, yes, but 
 - Card is special in the sense that it's automatically creating an HTML anchor based on the ID. So it has a nice, compact configuration surface, easy to interface with. The generic interface would be be different, not immediately usable with Card.
 - Our UI is mostly divided to Cards, so when we want to scroll somewhere, there is a good chance that it will be a card.

Reply:
Re-implemented as a hook, independent of card. (And then re-created ScrollingCard using the hook.)